### DISCUSSION TITLE
Is there any way to consume `unist-util-visit` from commonjs module synchronously

### DISCUSSION THREAD

**Discussion by maxkoretskyi** [syntax-tree/unist#97] [Q&A] [answered]
I'm stuck with a webpack config that is commonjs and doesn't allow async module load. I need to use [unist-util-visit](https://github.com/syntax-tree/unist-util-visit) library. What are my options? Maybe somehow build the library locally into commonjs?

**Comment by remcohaszing** [comment]
If you really need to use CJS, you can use for example esbuild to build `unist-util-visit` and its dependencies into a CJS bundle.

However, I would like to point out that Webpack supports ESM configuration files. I recommend to use that instead if possible.

**Comment by maxkoretskyi** [comment, marked as answer]
> If you really need to use CJS, you can use for example esbuild to build unist-util-visit and its dependencies into a CJS bundle.

thanks! is there a guide on how to do that?

> However, I would like to point out that Webpack supports ESM configuration files. I recommend to use that instead if possible.

yeah, I know, but my webpack config extends cjs webpack config that I don't have access, so I can't have part cjs and part esm

**Reply by wooorm** [reply]
```sh
npx esbuild unist-util-visit --bundle --platform=node --outfile=vendor/unist-util-visit.cjs
```

**Reply by maxkoretskyi** [reply]
appreciate very much!
